replica/database.cc

```diff
@@ -1660,7 +1660,7 @@ query::max_result_size database::get_query_max_result_size() const {
     switch (classify_request(_dbcfg)) {
         case request_class::user:
             return query::max_result_size(_cfg.max_memory_for_unlimited_query_soft_limit(), _cfg.max_memory_for_unlimited_query_hard_limit(),
-                    query::result_memory_limiter::maximum_result_size);
+                    _cfg.query_page_size_in_bytes());
```

Comment:
This is run on the replica, yes? So it can't be changed on a live cluster without triggering a read repair storm.

Options:
 - sample and pass this from the coordinator
 - rename to option to make it clear it's internal and can't be changed without dire effects

Reply:
This method is called from `storage_proxy` on the coordinator. These limits will then be sent to all replicas that participate in the read.

Reply:
good, so this is actually useful outside the test